lib/crystalball/map_generator/simple_map.rb

```diff
@@ -4,30 +4,57 @@ module Crystalball
   class MapGenerator
     # Basic map object for storing execution maps to storage
     class SimpleMap
-      def initialize(storage)
-        @storage = storage
-        @raw_map = {}
+      extend Forwardable
+
+      # Simple data object for map metadata
+      class Metadata
+        attr_accessor :commit, :type
+
+        def initialize(commit: nil, type: nil)
+          @commit = commit
+          @type = type
+        end
+
+        def to_h
+          {type: type, commit: commit}
+        end
       end
 
-      def load
-        self.raw_map = storage.load
+      attr_reader :cases, :metadata
+
+      delegate %i[commit commit=] => :metadata
+
+      def initialize(storage, metadata: nil, cases: {})
+        @storage = storage
+        @cases = cases
+        @metadata = Metadata.new(metadata || {type: self.class.name})
       end
 
       def stash(case_map)
-        raw_map[case_map.case_uid] = case_map.coverage
+        cases[case_map.case_uid] = case_map.coverage
       end
 
       def dump
```

Comment:
This `dump` seems outside of scope of `Map` class which aside from this method is just a data class. I get that it's needed for `StandardMap`, but I think we've been thinking about it wrong. Instead of `SimpleMap` and `StandardMap`, we should actually have one `Map` class, and `SimpleMapGenerator` and `StandardMapGenerator`.

Reply:
I think we need separate ticket for that. I'll create one